### ISSUE TITLE
querify doesn't encode single quotes correctly

### ISSUE THREAD

**Issue by ys** [gohugoio/hugo#11695]
### Does this issue reproduce with the latest release? Yes

When doing `querify "lol" .Summary` on a page with a `summary: Mon labo me le disait, y’a quelque chose de bizarre dans tes expositions.`

The encoded string is `lol=Mon+labo+me+le+disait%2C+y%26rsquo%3Ba+quelque+chose+de+bizarre+dans+tes+expositions.`

`%26rsquo%3B` which is `&rsquo;` but should be `%27`. Which is the correct way to URL encode apostrophe or single quote I think.

**Comment by jmooring** [comment]
On this page, how is the summary [defined](https://gohugo.io/content-management/summaries/#summary-splitting-options)? Automatically, manually (using `<!--more-->`), or in front matter?

**Comment by ys** [comment]
Manually as you can check on https://github.com/ys/bonjour/blob/main/content/posts/revision-leica/index.md?plain=1

**Comment by ys** [comment]
Usage of the querify can be seen in https://github.com/ys/bonjour/blob/main/themes/simple/layouts/partials/opengraph.html#L9

**Comment by ys** [comment]
I just tested `url.Values` `encode` method on https://pkg.go.dev/net/url#Values.Encode and it correctly encode it to `%27`

**Comment by jmooring** [comment]
The problem you are experiencing is closely related to <https://github.com/gohugoio/hugo/issues/8910#issuecomment-903158600>. Regardless of how you define the summary, it is passed through the markdown renderer, which converts certain punctuation characters to HTML entities. 

The `querify` function is behaving as expected/desired. For example:

```text
{{ querify "a" "b'c" }} --> a=b%27c 
```

#### Option 1

For a summary defined in front matter, you can access the raw value with:

```text
{{ querify "lol" .Params.Summary }}
```

But I'm not sure how forward-compatible that's going to be. See https://github.com/gohugoio/hugo/issues/11055.

#### Option 2

Change the character substitution rules in your site configuration:

```toml
[markup.goldmark.extensions.typographer]
apostrophe = "'"
leftSingleQuote = "'"
rightSingleQuote = "'"
```

#### Option 3

Disable the typographer extension:

```toml
[markup.goldmark.extensions.typographer]
disable = true
```

#### Option 4

Use a different front matter field, such as `description`:

```text
{{ querify "lol" .Description }}
```

To summarize, the `querify` function encodes single quotes correctly. The problem is that your code is feeding it an HTML entity.

**Comment by ys** [comment]
Wow, thanks so much, it makes a lot more sense then... Will pick an option. 
Thanks for the many ones!

Glad it is me and not the code...

**Comment by ys** [comment]
It looks like that `.Description` suffers a similar fate... it ends up as `%26%2339%3B`

**Comment by ys** [comment]
Nevermind, it is because I use `truncate`...

**Comment by jmooring** [comment]
```
{{ querify "a" ("b'c" | safeHTML | truncate 42) }}
```

<https://gohugo.io/functions/strings/truncate/>
